java/com/google/flatbuffers/Table.java

```diff
@@ -69,7 +69,8 @@ protected int __offset(int vtable_offset) {
 
   protected static int __offset(int vtable_offset, int offset, ByteBuffer bb) {
     int vtable = bb.array().length - offset;
-    return bb.getShort(vtable + vtable_offset - bb.getInt(vtable)) + vtable;
+    return bb.getShort(vtable + vtable_offset - bb.getShort(vtable)) != 0  ?
```

Comment:
But note you read the vtable offset with getShort.. whereas it is an int. Again, the code should correspond to what the __offset function above it does.